### DISCUSSION TITLE
ld -r -b binary cat.png -o cat.o

### DISCUSSION THREAD

**Discussion by trsh** [emscripten-core/emscripten#18319] [Q&A]
Can we do this with emcc?

I am referencing: https://csl.name/post/embedding-binary-data/

**Comment by sbc100** [comment]
That particular method is not currently supported by wasm-ld, no.   The version that uses assembly language does work though. For an example of how to do that see the package manager: https://github.com/emscripten-core/emscripten/blob/c93b3a109583b1cc6af83ab88b95f257eab21efc/tools/file_packager.py#L278-L323

Its conceivable we could add support to wasm-ld.. but the .s file is probably a better solution for now.

**Reply by sbc100** [reply]
Actually it looks like `lld` for ELF doesn't support `-b binary` either, so its unlikely it would make sense to add this to `wasm-ld`

**Reply by trsh** [reply]
Thank you. But I don't really understand the .py script.
